### PR TITLE
Fill opflash firsttime

### PR DESCRIPTION
**Pull request by jedori0228**
Filling `caf::SROpFlash::firsttime` as the first peaktime of the associated OpHit; copied from [icaruscode/Filters/FilterCRTPMTMatching_module.cc#349](https://github.com/SBNSoftware/icaruscode/blob/eff5062ee5f8c61559b505b000511e9f2e383efa/icaruscode/Filters/FilterCRTPMTMatching_module.cc#L349). The "trigger requirement" (https://github.com/SBNSoftware/icaruscode/blob/develop/icaruscode/Filters/FilterCRTPMTMatching_module.cc#L375) part is not available yet; how do we want this to be implemented to caf?

### CODE REVIEW

**Review by PetrilloAtWork — COMMENTED** [review]
It seems like GitHub did not take my review closure. Let's try again.

All good, some technical notes dropped around, **but**...

Given the purpose of the addition, I think it should be discussed whether to use the hit peak time, or start time, or even the rise time (which ICARUS has not enabled yet).
It may be good to discuss that on the coming optical reconstruction meeting, in fact (note for @amenegol too).

So I don't approve yet.

File: sbncode/CAFMaker/CAFMaker_module.cc [hunk 1]
```diff
@@ -1319,9 +1319,18 @@ void CAFMaker::produce(art::Event& evt) noexcept {
     if (flashes_handle.isValid()) {
       const std::vector<recob::OpFlash> &opflashes = *flashes_handle;
       int cryostat = ( pandora_tag_suffix.find("W") != std::string::npos ) ? 1 : 0;
+
+      // get associated OpHits for each OpFlash
+      art::FindManyP<recob::OpHit> findManyHits(flashes_handle, evt, fParams.OpFlashLabel() + pandora_tag_suffix);
```
[the remark this answers is not in the document]
Comment (jedori0228) [reply sbncode/CAFMaker/CAFMaker_module.cc]: Updated

File: sbncode/CAFMaker/CAFMaker_module.cc [hunk 2]
```diff
@@ -1319,9 +1319,18 @@ void CAFMaker::produce(art::Event& evt) noexcept {
...
+
+      int iflash=0;
       for (const recob::OpFlash& flash : opflashes) {
+
+        std::vector<art::Ptr<recob::OpHit>> ophits = findManyHits.at(iflash);
```
[the remark this answers is not in the document]
Comment (jedori0228) [reply sbncode/CAFMaker/CAFMaker_module.cc]: Updated

File: sbncode/CAFMaker/FillReco.cxx [hunk 3]
```diff
@@ -121,6 +122,14 @@ namespace caf
 
     srflash.time = flash.Time();
     srflash.timewidth = flash.TimeWidth();
+
+    double firstTime = 999999;
```
[the remark this answers is not in the document]
Comment (jedori0228) [reply sbncode/CAFMaker/FillReco.cxx]: Updated

File: sbncode/CAFMaker/FillReco.cxx [hunk 4]
```diff
@@ -121,6 +122,14 @@ namespace caf
...
+    for(const auto& hit: hits){
+      if (firstTime > hit->PeakTime())
+        firstTime = hit->PeakTime();
```
[the remark this answers is not in the document]
Comment (jedori0228) [reply sbncode/CAFMaker/FillReco.cxx]: Updated

**Review by francescopoppi — APPROVED** [review]
I think it looks on my side.

**Comment by PetrilloAtWork** [comment]
Thank you, @jedori0228: we are technically ok.

**Pull request merged**